### DISCUSSION TITLE
Using authorization_endpoint with oidc Passport Strategy

### DISCUSSION THREAD

**Discussion by IainMcHugh** [panva/openid-client#576] [Q&A] [answered]
Hello,

I have the following client configured (simplified for sharing):
```ts
const clientMetaData: ClientMetadata = {
    client_id: '...',
    client_secret: '...',
  };
const issuerMetaData: IssuerMetaData = {
    issuer: '...',
    authorization_endpoint: '...',
    registration_endpoint: '...',
    token_endpoint: ...',
    userinfo_endpoint: ...',
    jwks_uri: ...',
};
const params: AuthorizationParameters = {
    scope: '...',
    response_type: '...',
    redirect_uri: '...',
  };
const { Client } = new Issuer(issuerMetaData);
const client = new Client(clientMetaData);

const options: StrategyOptions = {
    client,
    params,
};

passport.use('oidc', new Strategy(options, verify));
// passport.serializeUser and deserializeUser configured also

authRouter.get('/authenticate', passport.authenticate('oidc'));
authRouter.get('/register', /** ??? */);
authRouter.get('/callback', passport.authenticate('oidc'), callbackHandler);
```

My login authentication flow works as expected as passport is handling PKCE stuff for me. Do you know is there a way to call a passport method that will invoke the `registration_endpoint` that is defined in the oidc Strategy options? I can see there's a [Client.register()](https://github.com/panva/node-openid-client/blob/main/docs/README.md#clientregistermetadata-other) but not sure how it could be used here.

At the moment I am manually generating the registration url with required query params (code_challenge, code_challenge_method etc) but the callback is failing as I think I'm missing steps that passport otherwise handles in the `/authenticate` flow

**Comment by panva** [comment, marked as answer]
I don't think you're clear what the registration functionality is for. It's not for end users, but for dynamically registering clients.

**Reply by IainMcHugh** [reply]
OK, I was half thinking that registration was being used in a different context when I was reading through the documentation but wasn't sure, thanks!
